# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/components/sessionstore/test/browser_firefoxView_selected_restore.js
# Commit: d29c1fad7862
# Full Hash: d29c1fad7862d61ef7c66d5989caa5ed0af4f011
# Author: Mike Conley <mconley@mozilla.com>
# Date: 2023-03-03 21:37:37
# Regressor Bug: 1423125
# File Overlap Count: 1
# Description:
#   Bug 1423125 - Restore previous session if undoCloseTab is called with no more tabs to restore. r=dao
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171377
# ==============================================================================

diff -r c81bb857ccff -r d29c1fad7862 browser/components/sessionstore/test/browser_firefoxView_selected_restore.js
--- a/browser/components/sessionstore/test/browser_firefoxView_selected_restore.js	Fri Mar 03 14:06:07 2023 +0000
+++ b/browser/components/sessionstore/test/browser_firefoxView_selected_restore.js	Fri Mar 03 14:06:43 2023 +0000
@@ -24,29 +24,6 @@
   ],
 };
 
-/**
- * This is regrettable, but when `promiseBrowserState` resolves, we're still
- * midway through loading the tabs. To avoid race conditions in URLs for tabs
- * being available, wait for all the loads to finish:
- */
-function promiseSessionStoreLoads(numberOfLoads) {
-  let loadsSeen = 0;
-  return new Promise(resolve => {
-    Services.obs.addObserver(function obs(browser) {
-      loadsSeen++;
-      if (loadsSeen == numberOfLoads) {
-        resolve();
-      }
-      // The typeof check is here to avoid one test messing with everything else by
-      // keeping the observer indefinitely.
-      if (typeof info == "undefined" || loadsSeen >= numberOfLoads) {
-        Services.obs.removeObserver(obs, "sessionstore-debug-tab-restored");
-      }
-      info("Saw load for " + browser.currentURI.spec);
-    }, "sessionstore-debug-tab-restored");
-  });
-}
-
 add_task(async function test_firefox_view_selected_tab() {
   let fxViewBtn = document.getElementById("firefox-view-button");
   ok(fxViewBtn, "Got the Firefox View button");